### ISSUE TITLE
Page iterator performs the first request twice

### ISSUE THREAD

**Issue by plamber** [microsoftgraph/msgraph-sdk-javascript#573]
# Bug Report
We are using PageIterator on our applications. Lately, we noticed that the PageIterator is performing two times the same call when being used. Ideally, the calls should be performed only once. I am using the latest version of the SDK.

## Steps to Reproduce
You can take the example on this official documentation: https://docs.microsoft.com/en-us/graph/sdks/paging?tabs=typeScript

- When executing this call you are getting back a PageCollection object. The call to the Graph API is performed
let response: PageCollection = await client
  .api('/me/messages?$top=10&$select=sender,subject')
  .get();
- We are passing the response object to the PageIterator
let pageIterator = new PageIterator(client, response, callback);
- When calling await pageIterator.iterate(); it is executing the very first call to Graph again instead of reusing the results from the first response. 

In a nutshell, it executes the same call twice. I would expect that we are reusing what is being passed in the constructor performing the first call only once. 

Let me know if you require any other information.

Cheers,
Patrick

**Comment by plamber** [comment]
Apology. Indeed, this was a misinterpretation and test from my end. Today I was able to identify that the problem lay in the way I was handling the callback function and resume operations.

Closing the issue now because it is not caused by the PageIterator

Thank you

**Comment by sebastienlevert** [comment]
Thanks for letting us know @plamber! Excited that you were able to find the issue! Do you feel this is a miss from our docs or it was more a mistake from your side?

**Comment by plamber** [comment]
Hi @sebastienlevert,
this time the mistake was more from my side. The docs were clear enough.

Cheers
